tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/test/utils/AssertionUtils.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.ballerinalang.testerina.test.utils;
+
+import org.ballerinalang.test.context.BallerinaTestException;
+
+/**
+ * Util class for test assertions.
+ */
+public class AssertionUtils {
+
+    public static void assertForTestFailures(String programOutput, String errMessage) throws BallerinaTestException {
+        if (programOutput.contains("error: there are test failures")) {
```

Comment:
Say hypothetically, a test failed due to a bad-sad or npe exception. 

Would the test automatically fail? Or will the `balClient` still return the buffered output?

Reply:
If a ballerina test fails it will anyway have the mentioned text in the program output. But, if it fails due to a compilation error, this fails to capture that. So, I'll add an else statement to capture unforeseen errors.

Reply:
I think the else statement seems to be causing the normal execution of tests to fail. Since if it can't find `error: there are test failures` it will automatically throw an exception.

Reply:
yeah will revert the else, and add a 'else if' to capture compilation issues in bal file. 
Every other test failure will be handled with the existing output.